cmd/util/api.go

```diff
@@ -1,35 +1,107 @@
 package util
 
 import (
-	"context"
+	"fmt"
 
 	"github.com/rs/zerolog/log"
 	"github.com/spf13/cobra"
 
 	"github.com/bacalhau-project/bacalhau/cmd/util/auth"
 	"github.com/bacalhau-project/bacalhau/pkg/config"
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
 	"github.com/bacalhau-project/bacalhau/pkg/publicapi/apimodels"
 	"github.com/bacalhau-project/bacalhau/pkg/publicapi/client"
 	clientv2 "github.com/bacalhau-project/bacalhau/pkg/publicapi/client/v2"
+	"github.com/bacalhau-project/bacalhau/pkg/setup"
 	"github.com/bacalhau-project/bacalhau/pkg/version"
 )
 
-func GetAPIClient(ctx context.Context) *client.APIClient {
-	legacyTLS := client.LegacyTLSSupport(config.ClientTLSConfig())
-	apiClient := client.NewAPIClient(legacyTLS, config.ClientAPIHost(), config.ClientAPIPort())
+func GetAPIClient(c config.Context) (*client.APIClient, error) {
+	repoDir, err := c.RepoPath()
+	if err != nil {
+		return nil, err
+	}
+	// TODO(forrest) [refactor]: a repo is required here as it contains the clients
+	// private key and a pk is needed in order to perform the authentication flow
+	// iff the configured authentication mode is challenge. We could decide to only
+	// require a repo here if the client _wants_ to communicate with a requester
+	// that uses authorization. In the event a requester doesn't need authorization
+	// we could use a un-authenticated client and not open/init a repo here.
```

Comment:
I would prefer to make the repo responsible for managing client keys. Sensitive items should not be stored in the config. This TODO lays out the intention for this change.